dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1801,6 +1778,79 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
+        List<String> candidates = chain.getCandidates();
+        ConflictRecord result = new ConflictRecord();
+        result.conflict = candidates.get(0);
+        boolean isEeCompatibleConflict = isEeCompatible(result.conflict);
+        if (chain.getChain().isEmpty()) {
+            // this is a configured root
+            result.configured = result.conflict;
+            result.chain = result.conflict;
+            if (isEeCompatibleConflict) {
+                // Note that this case should never happen because the compatible features are private
+                // they should never be allowed to be a configured root feature.
+                result.compatibleConflict = result.conflict;
+            }
+        } else {
+            result.configured = chain.getChain().get(0);
+            if (isEeCompatibleConflict) {
+                // Depending on how the feature resolver processes the included features
+                // it may not report the direct dependency on the compatible feature as the first
+                // conflict.  This may result in a reported chain that has more than one link to the
+                // compatible conflict.
+
+                // Check each level of the chain to see if it has a direct dependency on the compatible feature
+                // and is in conflict with the other chains in conflict.  If so then then use the feature
+                // in this level as the compatible conflict feature.
+                chainCheck: for (String feature : chain.getChain()) {
+                    ProvisioningFeatureDefinition featureDef = featureRepository.getFeature(feature);
+                    for (FeatureResource fr : featureDef.getConstituents(SubsystemContentType.FEATURE_TYPE)) {
+                        if (isCompatibleInConflictWithChains(fr, chain, inConflict)) {
+                            // this level is in conflict with the other chains; use it as the compatible conflict.
+                            result.compatibleConflict = feature;
+                            break chainCheck;
+                        }
+                    }
+                }
+                if (result.compatibleConflict == null) {
+                    // fall back to the second to last in chain that has a direct dependency on the compatible feature
+                    result.compatibleConflict = chain.getChain().get(chain.getChain().size() - 1);
+                }
+            }
+            result.chain = buildChainString(chain.getChain(), result.conflict);
+        }
+        return result;
+    }
+
+    private boolean isCompatibleInConflictWithChains(FeatureResource fr, Chain chain, Collection<Chain> inConflict) {
+        if (isEeCompatible(fr.getSymbolicName()) == false) {
+            // this included feature is not a compatible feature; move on to next
+            return false;
+        }
+
+        List<String> tolerates = fr.getTolerates();
```

Comment:
You can move the getTolerates() to line 1840.

Reply:
It could, but I was assigning it once to be used multiple times within each iteration of the for loop over the inConflict chain.

Reply:
I do not plan to address this.